brave/brave-core
# Adds grant claim process to Rewards Panel

Fixes: https://github.com/brave/brave-browser/issues/1783

<img width="406" alt="screen shot 2018-12-13 at 2 11 08 pm" src="https://user-images.githubusercontent.com/8732757/49968089-03865780-fee2-11e8-98cd-8d93a98ce75c.png">
<img width="385" alt="screen shot 2018-12-13 at 2 11 21 pm" src="https://user-images.githubusercontent.com/8732757/49968091-05e8b180-fee2-11e8-8d6d-0c03828ec4b5.png">
<img width="380" alt="screen shot 2018-12-13 at 2 11 32 pm" src="https://user-images.githubusercontent.com/8732757/49968094-07b27500-fee2-11e8-8381-8f7eb9e49005.png">


(See https://github.com/brave/brave-ui/pull/306#issue-236094082 for comments regarding design)

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on

## Test Plan:
1. Build Brave from this branch and run with a fresh profile
2. Navigate to a website such as `brave.com`
3. Click the `claim` button in the grant notification
4. Test to ensure the captcha fails appropriately
5. Test that the captcha can be dismissed and re-opened
6. Test to ensure the captcha can be solved appropriately
7. Test to ensure that captcha success notification can be dismissed
8. Ensure funds are added to balance as expected.

## Reviewer Checklist:
